### PR TITLE
RELATED: FET-487 Use more specific types for visualizationProperties

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**
Use more specific types for VisualizationProperties to minimize risk of missusing

---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [ ] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Successful `extended test - examples`
- [ ] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/3242
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2967

# Related Jira tasks
- FET-487: https://jira.intgdc.com/browse/FET-487

### CODE REVIEW

**Comment by ivan-nejezchleb** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1704.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7995/ : SUCCESS in 5m 58s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4208/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4209/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3558/ : SUCCESS in 18m 12s

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/4181/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4211/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3559/ : UNSTABLE in 10m 44s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7998/ : SUCCESS in 7m 06s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4212/ : SUCCESS in 3s

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7999/ : SUCCESS in 6m 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4213/ : SUCCESS in 4s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/4182/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4214/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3560/ : SUCCESS in 14m 47s

File: src/internal/interfaces/Visualization.ts [hunk 1]
```diff
@@ -261,11 +261,20 @@ export interface IUiConfig {
 }
 
 export interface IVisualizationProperties {
+    controls?: IVisualizationPropertiesControls;
+    sortItems?: AFM.SortItem[];
+}
+
+export interface IVisualizationPropertiesControls {
     // This can be anything depending on a visualization type
-    // perhaps consider adding: sortItems?: AFM.SortItem[]
```
Comment (ivanmjartan) [inline src/internal/interfaces/Visualization.ts:265]: Is this comment style valid ?

Comment (ivanmjartan) [reply src/internal/interfaces/Visualization.ts:265]: do we need type for this? Maybe any will be better.

File: src/internal/utils/sort.ts [hunk 2]
```diff
@@ -80,6 +80,7 @@ export function createSorts(
     const sortItems = get(visualizationProperties, "sortItems", []) as AFM.SortItem[];
     const sanitizedSortItems: AFM.SortItem[] = sanitizeSorts(afm, sortItems);
 
+    // TODO this function is never called from PivotTable
```
Comment (ivanmjartan) [inline src/internal/utils/sort.ts:83]: remove TODO

File: src/internal/interfaces/Visualization.ts (hunk 1 above)
Comment (ivan-nejezchleb) [reply src/internal/interfaces/Visualization.ts:265]: this comment is not valid, that is the reason why it is removed ;)

File: src/internal/utils/sort.ts (hunk 2 above)
Comment (ivan-nejezchleb) [reply src/internal/utils/sort.ts]: done

File: src/internal/interfaces/Visualization.ts (hunk 1 above)
Comment (ivanmjartan) [reply src/internal/interfaces/Visualization.ts:265]: sorry my fail was removed ok

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/8000/ : SUCCESS in 5m 47s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4215/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/8001/ : SUCCESS in 5m 52s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4216/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/8002/ : SUCCESS in 5m 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4217/ : SUCCESS in 3s

**Comment by ivan-nejezchleb** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1704.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4218/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3561/ : SUCCESS in 17m 07s

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/4183/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4219/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3562/ : SUCCESS in 11m 43s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/4184/ : SUCCESS in 10m 36s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-fossa-scanning-zuul-el7/162/ : SUCCESS in 21s (non-voting)
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/8003/ : SUCCESS in 2m 59s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1940/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2920/ : SUCCESS in 9m 10s

**Pull request merged**